File: backend/src/main/java/com/festago/dto/MemberTicketResponse.java
package com.festago.dto;

import com.festago.domain.EntryState;
import com.festago.domain.MemberTicket;
import com.festago.domain.Stage;
import java.time.LocalDateTime;

public record MemberTicketResponse(
    Long id,
    Integer number,
    LocalDateTime entryTime,
    EntryState state,
    LocalDateTime reservedAt,
    StageResponse stage,
    MemberTicketFestivalResponse festival) {

    private static final MemberTicketResponse EMPTY = new MemberTicketResponse(-1L, null, null, null, null, null, null);

    public static MemberTicketResponse from(MemberTicket memberTicket) {
        Stage stage = memberTicket.getStage();
        return new MemberTicketResponse(
            memberTicket.getId(),
            memberTicket.getNumber(),
            memberTicket.getEntryTime(),
            memberTicket.getEntryState(),
            memberTicket.getCreatedAt(),
            StageResponse.from(stage),
            MemberTicketFestivalResponse.from(stage.getFestival()));
    }

    public static MemberTicketResponse empty() {
        return EMPTY;
    }
}

File: android/festago/app/src/main/java/com/festago/festago/data/dto/MemberTicketResponse.kt
package com.festago.festago.data.dto

import com.festago.festago.model.Ticket
import com.festago.festago.model.TicketCondition
import kotlinx.serialization.SerialName
import kotlinx.serialization.Serializable
import java.time.LocalDateTime

@Serializable
data class MemberTicketResponse(
    val id: Long,
    val number: Int,
    val entryTime: String,
    @SerialName("state") val condition: String,
    val reservedAt: String,
    val stage: StageResponse,
    val festival: MemberTicketFestivalResponse,
) {
    fun toDomain(): Ticket = Ticket(
        id = id,
        number = number,
        entryTime = LocalDateTime.parse(entryTime),
        condition = stateToDomain(condition),
        reserveAt = LocalDateTime.parse(reservedAt),
        stage = stage.toDomain(),
        festivalTicket = festival.toDomain(),
    )

    private fun stateToDomain(state: String): TicketCondition {
        return when (state) {
            "BEFORE_ENTRY" -> TicketCondition.BEFORE_ENTRY
            "AFTER_ENTRY" -> TicketCondition.AFTER_ENTRY
            "AWAY" -> TicketCondition.AWAY
            else -> throw IllegalArgumentException("Unknown ticket state: $state")
        }
    }
}
